src/lovok_handle_internal.cpp

```diff
@@ -67,8 +67,14 @@ bool SubtractUnderflow(uint64_t a, uint64_t b, uint64_t *result) {
     return true;
 }
 
-LovokStatusCode ParseBoxes(FileWrapper *fileWrapper, uint64_t length, uint64_t byteOffset, const std::function<LovokStatusCode(const Box &, uint64_t)> &f) {
-    if (length < 0) { return PARSE_ERROR; }
+LovokStatusCode ParseBoxes(FileWrapper *fileWrapper, uint64_t length, uint64_t byteOffset, uint64_t offset, const std::function<LovokStatusCode(const Box &, uint64_t)> &f) {
```

Comment:
I think the presence of both `byteOffset` and `offset` is confusing.  Can you explain that?

Reply:
`byteOffset` is the current `offset`. I think `offset` here is the value to be added to `byteOffset` and to be subtracted from `length`. Should we rename it to `headerSize`? The idea of this PR is to move the logic to parseHeader to add to `byteOffset` and subtract from `length` that way we can have one place where the value is computed and checked for overflow and underflow.

Reply:
I think a rename is fair.   `headerSize` seems like a good name.